### DISCUSSION TITLE
Scaling the content to fit window size

### DISCUSSION THREAD

**Discussion by jesnor** [emilk/egui#1217] [Q&A]
Hello, thanks for a great library that's really easy to use :)

I'm using egui for a VST plugin and want to support user window resizing, however I still want the exact same widget layout regardless of window size and instead set the pixels per point scaling factor depending on how much space is available. I've experimented a bit but haven't really found a good way to achieve that. Any tips?

**Comment by emilk** [comment]
Changing `pixels_per_point` is the right approach, but how would you handle a size with the wrong aspect ratio?

**Reply by jesnor** [reply]
> Changing `pixels_per_point` is the right approach, but how would you handle a size with the wrong aspect ratio?

I will scale to the minimum of available width and height while keeping aspect ratio. The remaining window space will be blank.
